src/library/Dropdown/Dropdown.js

```diff
@@ -114,43 +153,65 @@ export default class Dropdown extends Component<Props, State> {
   }
 
   render() {
-    const {
-      children,
-      data,
-      modifiers,
-      placement,
-      wide,
-      ...restProps
-    } = this.props;
+    const { data: ignoreData, ...restProps } = this.props;
     const isOpen = this.getControllableValue('isOpen');
-
-    const dropdownContentProps = {
-      data,
-      id: this.getContentId(),
-      getItemProps: this.getItemProps,
-      getMenuProps: this.getMenuProps,
-      modifiers,
-      placement,
-      wide
-    };
+    const { renderTrigger } = this.props;
 
     const rootProps = {
-      id: this.id,
       ...restProps,
-      content: <DropdownContent {...dropdownContentProps} />,
-      getTriggerProps: this.getTriggerProps,
+      id: this.id,
       isOpen,
       onClose: this.close,
       onOpen: this.open,
-      triggerRef: (node: ?React$Component<*, *>) => {
-        this.dropdownTrigger = node;
-      },
-      wrapContent: false
+      renderContent: this.renderContent,
+      ...(renderTrigger ? { renderTrigger: this.renderTrigger } : {}),
+      triggerRef: this.setTriggerRef
     };
 
-    return <Root {...rootProps}>{children}</Root>;
+    return <Root {...rootProps}>{!renderTrigger && this.renderTrigger()}</Root>;
   }
 
+  getStateAndHelpers = (): StateAndHelpers => {
+    return {
+      state: {
+        highlightedIndex: this.getControllableValue('highlightedIndex'),
+        isOpen: this.getControllableValue('isOpen')
+      },
+      helpers: {
+        close: this.close,
+        focusTrigger: this.focusTrigger,
+        open: this.open
+      }
+    };
+  };
+
+  setTriggerRef = (node: ?React$Component<*, *>) => {
```

Comment:
Alpha-sort methods after `render()`. (Assuming this applies to all files)

Reply:
After reviewing further, I like that `get_____Props` is always right above `render_____`. Maybe alpha-sort methods, but treat those two as if they're just `_____`?

Reply:
Yeah, I don't want to alpha sort methods after render.  I often find that it is more useful to group them logically.

Reply:
Fair enough. I couldn't discern a common order-by-logic, though.